client/src/components/nav-bar.tsx

```diff
@@ -206,22 +215,42 @@ function NavMenu(props: {
     </List>
   );
 }
-
 export function NavBar(props: {
   mentorId: string | undefined;
   title: string;
   search: {
     back?: string;
   };
+  uploads: UploadTask[];
+  uploadsButtonVisible: boolean;
+  toggleUploadsButtonVisibility: (b: boolean) => void;
 }): JSX.Element {
   const classes = useStyles();
   const { back } = props.search;
-  const [isDrawerOpen, setIsDrawerOpen] = React.useState(false);
+  const {
+    uploads,
+    uploadsButtonVisible,
+    toggleUploadsButtonVisibility,
+  } = props;
+  const numUploadsInProgress = uploads?.filter(
+    (upload) =>
+      upload.uploadStatus !== UploadStatus.DONE &&
+      upload.uploadStatus !== UploadStatus.CANCELLED
+  ).length;
+  const numUploadsComplete = uploads?.filter(
+    (upload) => upload.uploadStatus == UploadStatus.DONE
+  ).length;
 
+  const [isDrawerOpen, setIsDrawerOpen] = React.useState(false);
+  //if there are no uploads, defaults to true, else if there are any uploads that aren't yet cancelled, should not be disabled
+  const disableUploadsButton = uploads
+    ? uploads.find(
```

Comment:
use falsyness...almost never want to check for `== undefined`

```typescript
? Boolean(uploads.find(...))
```

```

Reply:
fd9f53b